### PR TITLE
add 17 test cases for nip57

### PR DESCRIPTION
**Pull request by sepehr-safari**
getZapEndpoint
    √ returns null if neither lud06 nor lud16 is present (1 ms)
    √ returns null if fetch fails (1 ms)
    √ returns null if the response does not allow Nostr payments
    √ returns the callback URL if the response allows Nostr payments
  makeZapRequest
    √ throws an error if amount is not given (29 ms)
    √ throws an error if profile is not given
    √ returns a valid Zap request (1 ms)
  validateZapRequest
    √ returns an error message for invalid JSON
    √ returns an error message if the Zap request is not a valid Nostr event
    √ returns an error message if the signature on the Zap request is invalid (46 ms)
    √ returns an error message if the Zap request does not have a "p" tag (10 ms)
    √ returns an error message if the "p" tag on the Zap request is not valid hex (7 ms)
    √ returns an error message if the "e" tag on the Zap request is not valid hex (7 ms)
    √ returns an error message if the Zap request does not have a relays tag (10 ms)
    √ returns null for a valid Zap request (8 ms)
  makeZapReceipt
    √ returns a valid Zap receipt with a preimage (6 ms)
    √ returns a valid Zap receipt without a preimage (6 ms)

Test Suites: 1 passed, 1 total
Tests:       17 passed, 17 total

### CODE REVIEW

File: nip57.test.js [hunk 1]
```diff
@@ -0,0 +1,335 @@
+const {bech32} = require('@scure/base')
```
Comment (sepehr-safari) [inline nip57.test.js:1]: just forgot to remove this unnecessary import!
it can be safely removed.

**Comment by fiatjaf** [comment]
Thank you very much!

**Comment by fiatjaf** [comment]
Did you generate these tests using some AI bot?

**Comment by sepehr-safari** [comment]
> Thank you very much!

my pleasure.



> Did you generate these tests using some AI bot?

Nope.
I've seen your note regarding writing a bunch of test cases.
[https://www.nostribe.com/post/note19gx4944w3eewe4lkm5yrz528ntyz2y9hm9kxgtxwne5vdh9ng2gsyaueqv](https://www.nostribe.com/post/note19gx4944w3eewe4lkm5yrz528ntyz2y9hm9kxgtxwne5vdh9ng2gsyaueqv)
So, I started with client tools, writing these tests over each available method and their expected functionality, it's not a big deal.

**Pull request merged**